runner/operator/roslin_qc_operator/bin/input.py

```diff
@@ -0,0 +1,395 @@
+
+def pair_to_job_data(pair):
+    """
+    Convert a single tumor normal pair dict to Job submission data format
+    NOTE: not the same as CWL format
+
+    pair = {
+        'normal': {
+            'R1': FileMetadata,
+            'R2': FileMetadata
+        },
+        'tumor': {
+            'R1': FileMetadata,
+            'R2': FileMetadata
+        }
+    """
+    # get the base datastructure dict for each R1 and R2
+    normal_r1 = build_sample(pair['normal']['R1'])
+    normal_r2 = build_sample(pair['normal']['R2'])
+    tumor_r1 = build_sample(pair['tumor']['R1'])
+    tumor_r2 = build_sample(pair['tumor']['R2'])
+
+    # merge the dict's; the second dict will overwrite args from the first
+    normal = {**normal_r1, **normal_r2}
+    normal['R1'] = file_to_job_data(pair['normal']['R1'].file)
+    normal['R2'] = file_to_job_data(pair['normal']['R2'].file)
+
+    tumor = {**tumor_r1, **tumor_r2}
+    tumor['R1'] = file_to_job_data(pair['tumor']['R1'].file)
+    tumor['R2'] = file_to_job_data(pair['tumor']['R2'].file)
+
+    output = (tumor, normal)
+    return(output)
+
+def bams_to_cwl(bams):
+    """
+    Convert the items in the bams list to CWL format
+
+    bams = [ [File.objects.get(file_name = 's_C_K2902H_P001_d.rg.md.abra.printreads.bam'), File.objects.get(file_name = 's_C_K2902H_N001_d.rg.md.abra.printreads.bam')], ... ]
+    """
+    bam_cwls = []
+    for item in bams:
+        tumor_bam = item[0]
+        normal_bam = item[1]
+        tumor_bam_cwl = file_to_job_data(tumor_bam)
+        normal_bam_cwl = file_to_job_data(normal_bam)
+        bam_cwls.append([tumor_bam_cwl, normal_bam_cwl])
+    return(bam_cwls)
+
+def parse_runparams_from_ports(ports_queryset):
+    """
+    Get the runparams from a Port queryset
+    """
+    # TODO: how to ensure the correct number of Port items? Should be 1
+    param_port = ports_queryset.filter(name = "runparams").first()
+    runparams = {}
+    runparams['project_prefix'] = param_port.value['project_prefix'][0] # NOTE: comes in as array but need single string
+    runparams['genome'] = param_port.value['genome']
+    runparams['scripts_bin'] = param_port.value['scripts_bin']
+
+    # TODO: find a way to handle
+    runparams['pi'] = "NA"
+    runparams['pi_email'] = "NA"
+    return(runparams)
+
+def get_db_files(assay, references_json = "runner/operator/roslin_operator/reference_jsons/roslin_resources.json"):
+    """
+    Return a dict with the required reference file paths
+    """
+    references = json.load(open(references_json))
+    db_files = {}
+    db_files['fp_genotypes'] = references["targets"][assay]['FP_genotypes']
+    db_files['hotspot_list_maf'] = references["request_files"]['hotspot_list_maf']
+    db_files['conpair_markers'] = references["request_files"]['conpair_markers']
+    db_files['ref_fasta'] = references["request_files"]['ref_fasta']
+
+    # need to resolve some items to URI's
+    for key in ['fp_genotypes', 'hotspot_list_maf', 'ref_fasta']:
```

Comment:
Are these files specified in references_json `runner/operator/roslin_operator/reference_jsons/roslin_resources.json`? If they are why do we have this logic to check is it a path or uri?